test/test_nn.py

```diff
@@ -7608,5 +7612,30 @@ def __call__(self, input):
     fullname='AdaptiveLogSoftmax'))
 
 
+def parse_args():
+    parser = argparse.ArgumentParser(
+        description='Run only a portion of test_nn')
+    parser.add_argument(
+        '--shard',
+        type=int,
+        required=False,
+        help='the shard number to run')
+    parser.add_argument(
+        '--verbose',
+        action='store_true',
+        required=False,
+        help='verbose mode')
+    return parser.parse_args()
+
+
 if __name__ == '__main__':
+    options = parse_args()
+    if options.shard is not None:
+        def load_tests(loader, tests, pattern):
```

Comment:
How exactly does this connect to the rest of our test suite? Isn’t it just left here as a local function?

Reply:
Yeah it's just a function defined when `__name__ == '__main__'`, but it is the unitttest protocol. See https://docs.python.org/2/library/unittest.html#load-tests-protocol